modulemd/modulemd-eol.c

```diff
@@ -0,0 +1,1196 @@
+    }
+  return self->obsoleted_by_module_name;
+}
+
+
+const gchar *
+modulemd_eol_get_obsoleted_by_module_stream (ModulemdEol *self)
+{
+  /* It should not be possible to set reset and obsoleted_by*
+   * at the same time but to be sure check anyway */
+  if (self->reset)
+    {
+      return NULL;
+    }
+  return self->obsoleted_by_module_stream;
+}
+
+
+static void
+modulemd_eol_set_version (ModulemdEol *self, guint64 version)
+{
+  g_return_if_fail (MODULEMD_IS_EOL (self));
+  g_return_if_fail (version != 0);
+
+  self->version = version;
+
+  g_object_notify_by_pspec (G_OBJECT (self), properties[PROP_VERSION]);
+}
+
+void
+modulemd_eol_set_modified (ModulemdEol *self, guint64 modified)
+{
+  g_return_if_fail (MODULEMD_IS_EOL (self));
+
+  self->modified = modified;
+
+  g_object_notify_by_pspec (G_OBJECT (self), properties[PROP_MODIFIED]);
+}
+
+
+void
+modulemd_eol_set_reset (ModulemdEol *self, gboolean reset)
+{
+  g_return_if_fail (MODULEMD_IS_EOL (self));
+  g_return_if_fail (!(reset && self->obsoleted_by_module_name));
+  g_return_if_fail (!(reset && self->obsoleted_by_module_stream));
+  g_return_if_fail (!(reset && self->eol_date));
+
+  self->reset = reset;
+
+  g_object_notify_by_pspec (G_OBJECT (self), properties[PROP_RESET]);
+}
+
+
+static void
+modulemd_eol_set_module_name (ModulemdEol *self, const gchar *module_name)
+{
+  g_return_if_fail (MODULEMD_IS_EOL (self));
+  g_return_if_fail (module_name);
+  g_return_if_fail (g_strcmp0 (module_name, O_DEFAULT_STRING));
+
+  g_clear_pointer (&self->module_name, g_free);
+  self->module_name = g_strdup (module_name);
+
+  g_object_notify_by_pspec (G_OBJECT (self), properties[PROP_MODULE_NAME]);
+}
+
+
+static void
+modulemd_eol_set_module_stream (ModulemdEol *self, const gchar *module_stream)
+{
+  g_return_if_fail (MODULEMD_IS_EOL (self));
+  g_return_if_fail (module_stream);
+  g_return_if_fail (g_strcmp0 (module_stream, O_DEFAULT_STRING));
+
+  g_clear_pointer (&self->module_stream, g_free);
+  self->module_stream = g_strdup (module_stream);
+
+  g_object_notify_by_pspec (G_OBJECT (self), properties[PROP_MODULE_STREAM]);
+}
+
+
+void
+modulemd_eol_set_module_context (ModulemdEol *self,
+                                 const gchar *module_context)
+{
+  g_return_if_fail (MODULEMD_IS_EOL (self));
+
+  g_clear_pointer (&self->module_context, g_free);
+  self->module_context = g_strdup (module_context);
+
+  g_object_notify_by_pspec (G_OBJECT (self), properties[PROP_MODULE_CONTEXT]);
+}
+
+void
+modulemd_eol_set_eol_date (ModulemdEol *self, guint64 eol_date)
+{
+  g_return_if_fail (MODULEMD_IS_EOL (self));
+  g_return_if_fail (!(eol_date && self->reset));
+
+  self->eol_date = eol_date;
+
+  g_object_notify_by_pspec (G_OBJECT (self), properties[PROP_EOL_DATE]);
+}
+
+
+void
+modulemd_eol_set_message (ModulemdEol *self, const gchar *message)
+{
+  g_return_if_fail (MODULEMD_IS_EOL (self));
+  g_return_if_fail (message);
+  g_return_if_fail (g_strcmp0 (message, O_DEFAULT_STRING));
```

Comment:
Don't attempt to validate this here. It will cause problems for anyone attempting to initialize on instantiation. make this a check in `modulemd_eol_validate()`.